packages/patternlab/styleguide/source/assets/js/modules/mainNav.js

```diff
@@ -115,109 +59,123 @@ export default (function (window, document, $, undefined) {
         //  - Open previous pull down menu and select first item
         hide($openContent);
         $topLevelLink.attr('aria-expanded', 'false');
-        $link.removeClass(hasFocus);
+        $topLevelItem.removeClass(hasFocus);
         // Get previous item if left arrow, next item if right arrow.
         index += (prev ? -1 : 1);
         // Wrap around if at the end of the set of menus.
         index = ((index % linkCount) + linkCount) % linkCount;
-        $topLevelLinks[index].focus();
-        return;
+        $mainNavItemsToggle[index].focus();
+        // return;
       }
 
-    })
-      .on('mouseenter', function (e) {
-        $(this).children('button').attr("aria-expanded", "true");
-        $('.has-focus').removeClass('has-focus');
-
-        if (windowWidth > breakpoint) {
-          let $openContent = $(this).find('.js-main-nav-content');
-          show($openContent);
+      if (action.up || action.down) {
+        // If submenu is not already
+        if ($openContent.length === 0) {
+          // Open the submenu
+          show($dropdownContent);
+          $topLevelLink.attr('aria-expanded', 'true');
+          $topLevelItem.addClass(openClass);
+          if (action.up) {
+            console.log('up')
+            // key up focus on the last item
+            dropdownIndex = dropdownLinksLength - 1;
+          } else {
+            console.log('down')
+            // key down focus on the first item
+            dropdownIndex = 0;
+          }  
+          console.log(dropdownIndex)
+          console.log($dropdownLinks[dropdownIndex])
+          // not sure why this doesn't work?????
+          $dropdownLinks[dropdownIndex].focus(); 

```

Comment:
```suggestion
          $dropdownLinks[dropdownIndex].focus(); 

```
Unsure why this is not working. See the first up/down arrow keypress on the top level
![main-nav-keyboard](https://user-images.githubusercontent.com/5789411/230972972-6388f1c8-c44e-473a-b4e2-935e5aed364d.gif)